k8s/helm/prefect-agent/templates/deployment.yaml

```diff
@@ -0,0 +1,121 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Values.nameOverride | default .Chart.Name | trunc 63 | trimSuffix "-" }}
+  labels:
+    {{- include "prefect-agent.commonLabels" . | nindent 4 }}
+    app.kubernetes.io/component: agent
+  annotations:
+    {{- merge .Values.annotations .Values.annotations | toYaml | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicas }}
+  {{- with .Values.strategy }}
+  strategy:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "prefect-agent.matchLabels" . | nindent 6 }}
+      app.kubernetes.io/component: agent
+  template:
+    metadata:
+      labels:
+        {{- include "prefect-agent.matchLabels" . | nindent 8 }}
+        app.kubernetes.io/component: agent
+      {{- with .Values.podAnnotations }}
+      annotations: {{- toYaml . | nindent 8 }}
+      {{- end }}
+    spec:
+    {{- if .Values.image.pullSecrets }}
+      imagePullSecrets:
+        {{- toYaml .Values.image.pullSecrets | nindent 8 }}
+    {{- else if .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml .Values.imagePullSecrets | nindent 8 }}
+    {{- end }}
+      serviceAccountName: {{ include "prefect-agent.serviceAccountName" . }}
+      {{- with .Values.podSecurityContext }}
+      securityContext:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      containers:
+      - name: agent
+        {{- with .Values.podSecurityContext }}
+        securityContext:
+          {{- toYaml . | nindent 8 }}
+        {{- end }}
+        image: "{{ .Values.image.name }}:{{ .Values.image.tag | default .Values.prefectVersionTag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy  }}
+        command:
+          - bash
+          - "-c"
+          {{- if .Values.jobTemplateFilePath }}
+          - "prefect agent kubernetes start --job-template {{ .Values.jobTemplateFilePath }}"
+          {{- else }}
+          - "prefect agent kubernetes start"
+          {{- end }}
+        env:
+          - name: PREFECT__CLOUD__AGENT__AUTH_TOKEN
+            valueFrom:
+              secretKeyRef:
+                name: prefect-config
```

Comment:
allow for this to be the default, but be overridden if an existing secret name is provided